lib/Controller/PageController.php

```diff
@@ -89,6 +120,25 @@ public function index(): TemplateResponse {
 		$this->initialStateService->provideInitialState($this->appName, 'croppedLayout', $this->config->getUserValue($user->getUid(), Application::APP_ID, 'croppedLayout', 'false'));
 		$this->initialStateService->provideInitialState($this->appName, 'systemtags', $this->appManager->isEnabledForUser('systemtags') === true);
 
+		$paths = [];
+		try {
+			$userFolder = $this->rootFolder->getUserFolder($user->getUID());
+			$key = $user->getUID() . ':' . $userFolder->getEtag();
+			$paths = $this->nomediaPathsCache->get($key);
+			if ($paths === null) {
+				$nomedia = $userFolder->search('.nomedia');
+				$noimage = $userFolder->search('.noimage');
```

Comment:
You can search this in one go with

```
$userFolder->search(new SearchQuery(new SearchBinaryOperator(ISearchBinaryOperator::OR, [
    new SearchComparison(ISearchComparison::COMPARE_EQ, 'name', '.nomedia'),
    new SearchComparison(ISearchComparison::COMPARE_EQ, 'name', '.noimage')
]), 0, 0, [], $user));
```

that also makes it not use a wildcard needlessly.

Reply:
> SearchQuery SearchBinaryOperator SearchComparison

Are these in the OCP namespace, though?

Reply:
No, interfaces for them are though, but I never got around to making a query builder for it.

Reply:
@skjnldsv Are you ok with that?

Reply:
@skjnldsv I think this needs a simple 👍 or 👎

Reply:
Yes, let get this moving forward :)